app/src/main/java/org/simple/clinic/home/overdue/OverdueScreenController.kt

```diff
@@ -77,4 +84,27 @@ class OverdueScreenController @Inject constructor(
   private fun calculateOverdueDays(date: LocalDate): Int {
     return DAYS.between(date, LocalDate.now(ZoneOffset.UTC)).toInt()
   }
+
+  private fun phoneCallPermissionRequests(events: Observable<UiEvent>): Observable<UiChange> {
+    return events.ofType<CallPatientClicked>()
+        .map { { ui: Ui -> ui.requestCallPermission() } }
+  }
+
+  private fun patientCalls(events: Observable<UiEvent>): Observable<UiChange> {
+    val callPhonePermissionChanges = events
+        .ofType<CallPhonePermissionChanged>()
+        .map(CallPhonePermissionChanged::result)
+
+    val grantedStream = callPhonePermissionChanges
+        .filter { it == RuntimePermissionResult.GRANTED }
+        .withLatestFrom(events.ofType<CallPatientClicked>())
+        .flatMap { (_, callClicked) -> Observable.just { ui: Ui -> ui.callPatientWithoutUsingDialer(callClicked.phoneNumber) } }
```

Comment:
You don't need a `flatMap` here. 

```
.flatMap { (_, callClicked) -> { ui: Ui -> ui.callPatientWithoutUsingDialer(callClicked.phoneNumber) } }
```

Reply:
We can also extract `events.ofType<CallPatientClicked>()` to make `grantedStream` and `notGrantedStream` shorter.

```
val callPhonePermissionChanges = events
        .ofType<CallPhonePermissionChanged>()
        .map(CallPhonePermissionChanged::result)

val callClicks = events
        .ofType<CallPatientClicked>()
        .map { it.phoneNumber }

val grantedStream = callPhonePermissionChanges
        .filter { it == RuntimePermissionResult.GRANTED }
        .withLatestFrom(callClicks)
        .map { (_, number) -> { ui: Ui -> ui.callPatientWithoutUsingDialer(number) } }

val notGrantedStream = callPhonePermissionChanges
        .filter { it != RuntimePermissionResult.GRANTED }
        .withLatestFrom(callClicks)
        .flatMap { (_, number) -> { ui: Ui -> ui.callPatientUsingDialer(number) } }
```

Reply:
It looks like `grantedStream` and `notGrantedStream` names can be improved. We've so far named streams according to their jobs. 

For e.g., `updateListStream` in the `screenSetup()` is self descriptive that the stream updates the list. 

`grantedStream` in comparison does not indicate what it is doing. It only indicates that this stream will function when the permission is granted.